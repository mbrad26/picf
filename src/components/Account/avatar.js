import React, { useState, useEffect } from 'react';
import { Image } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { doGetAvatarUrl } from '../../redux/actions/user';

const Avatar = ({ uid }) => {
  const dispatch = useDispatch();
  const { avatarUrl } = useSelector(state => state.userState);
  const [state, setState] = useState(avatarUrl);

  useEffect(() => {
    dispatch(doGetAvatarUrl(uid));
    setState(avatarUrl);
  }, [dispatch, uid, avatarUrl]);

  return (
    <>
      <Image loading='eager' id='avatar' src={state} roundedCircle/>
    </>
  );
};

export default Avatar;
