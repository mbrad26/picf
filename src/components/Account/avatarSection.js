import React, { lazy, Suspense } from 'react';
import Badge from '@material-ui/core/Badge';

import Avatar from './avatar';
import AvatarUploadForm from '../UploadForm/avatarUploadForm';

const AvatarSection = () => {
  const authUser = JSON.parse(localStorage.getItem('authUser'));
  const uid = authUser.uid;

  return (
    <div>
      <h3>Profile</h3>
      <hr />
      <h5>Profile picture</h5>
      <Badge 
        color="secondary" 
        overlap="circle" 
        badgeContent={<label htmlFor="avatar-upload"><span>Edit Avatar</span></label>}
        anchorOrigin={{ vertical: 'top', horizontal: 'left' }}
      >
        <Avatar uid={uid} />
      </Badge>
      <AvatarUploadForm />
    </div>
  )
}

export default AvatarSection;
