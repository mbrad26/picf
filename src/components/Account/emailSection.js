import React,  { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { doUpdateEmailRequest } from '../../redux/actions/user';

const EmailSection = () => {
  const dispatch = useDispatch();
  const { authUser, authError } = useSelector(state => state.userState);
  const [state, setState] = useState({ email: authUser.email, error: authError});
  const { email, error } = state;

  const onSubmit = event => {
    event.preventDefault();
    dispatch(doUpdateEmailRequest(email));
  };

  const onChange = event => 
    setState({ ...state, email: event.target.value });

  return (
    <div className='section'>
      <h5>Email</h5>
      <form onSubmit={onSubmit}>
        <input 
          type='email'
          name='email'
          value={email}
          onChange={onChange}
          placeholder='Email'
        />
      <button type='submit'>Update email</button>

      {error && <p>{error.message}</p>}
    </form>
    </div>
  );
};

export default EmailSection;
