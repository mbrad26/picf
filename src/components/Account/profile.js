import React from 'react';

const Profile = () => {
  return (
    <div className='profile-container'>
      <h3>Profile</h3>
      <p>Avatar</p>
      <p>Username</p>
      <p>Email</p>
      <p>Details</p>
    </div>
  )
}

export default Profile;
