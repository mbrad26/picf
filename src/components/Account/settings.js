import React from 'react';

import PasswordChangeForm from '../PasswordChange';
import PasswordResetForm from '../PasswordForget/passwordResetForm';

const Settings = () => {
  return (
    <div className='settings-container'>
      <h3>Password reset</h3>
      <PasswordResetForm />
      <h3>Password change</h3>
      <PasswordChangeForm />
    </div>
  )
}

export default Settings;
