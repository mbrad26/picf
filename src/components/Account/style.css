.component-container {
  display: grid;
  grid-template-columns: 2fr 10fr;
}

.container .nav-link {
  color: black;
}

.settings-container {
  margin: 1em;
}

.profile-container {
  margin: 1em;
}