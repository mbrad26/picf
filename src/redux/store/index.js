import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from  'redux';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

export default store;