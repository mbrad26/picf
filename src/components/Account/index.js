import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Switch, Route, Redirect } from 'react-router-dom';

import './style.css';
import Profile from './profile';
import Sidebar from '../Sidebar';
import Gallery from '../Gallery';
import Settings from './settings';
import * as ROUTES from '../constants/routes';
import { doEmailVerificationRequest } from '../../redux/actions/user';

const Account = () => {
  const dispatch = useDispatch();
  const { authUser } = useSelector(state => state.userState);

  const onClick = () => dispatch(doEmailVerificationRequest());

  if(!authUser) return <Redirect to={ROUTES.SIGN_IN}/>

  if(authUser && 
    !authUser.emailVerified &&
    authUser.providerData
            .map(provider => provider.providerId)
            .includes('password')
  ) {
    return (
      <div>
        <p>
          Verify your E-Mail: Check your E-Mails (Spam folder 
          included) for a confirmation E-Mail or 
          send another confirmation E-Mail.
        </p>

        <button
          type="button" 
          onClick={onClick}
        >
          Send confirmation E-Mail
        </button>
      </div>
    );
  };

  return (
    <div className='component-container'>
      <div className='sidebar'>
        <Sidebar />
      </div>
      <div className='main'>
        <Switch>
          <Route exact path={ROUTES.ACCOUNT} component={Profile} />
          <Route path={ROUTES.SETTINGS} component={Settings} />
          <Route path={ROUTES.TIMELINE} component={Gallery} />
        </Switch>
      </div>
    </div>
  );
};

export default Account;
