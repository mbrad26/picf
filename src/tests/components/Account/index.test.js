import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import store from '../../../redux/store';
import Account from  '../../../components/Account';

describe('Account', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store} >
        <MemoryRouter initialEntries={['/account']}>
          <Account />
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});