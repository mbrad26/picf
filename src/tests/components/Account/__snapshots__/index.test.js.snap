// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders snapshot 1`] = `null`;
